### PR TITLE
Use vite over webpack

### PR DESCRIPTION
**Pull request by Mafii**
Partially solves #11 

* [x] Use vite (this allows using newer node versions and faster development)
* [x] Use newer versions of packages 
* [x] Re-add linters and git hooks
* [x] Decide if we want lint on run dev (otherwise drop the last commit again)
* [x] Verify everything works correctly

### CODE REVIEW

**Comment by StefanieJaeger** [comment]
If @jeremystucki is busy, maybe I could do the review?

File: README.md [hunk 1]
```diff
@@ -13,18 +13,23 @@ npm install
...
 npm run build
 ```
 
-### Lints and fixes files
+### Lints
 ```
 npm run lint
 ```
 
+### TODO: Lints and fixes files
```
Comment (StefanieJaeger) [inline README.md:29]: 👁️

File: package.json [hunk 2]
```diff
@@ -42,23 +36,13 @@
...
-  "gitHooks": {
-    "pre-commit": "lint-staged"
-  },
-  "lint-staged": {
-    "*.{js,jsx,vue}": [
-      "vue-cli-service lint",
-      "git add"
-    ]
-  }
+  "gitHooks": {},
```
[the remark this answers is not in the document]
Comment (StefanieJaeger) [reply package.json:46]: I discussed this with @jeremystucki. I think, we should remove the gitHooks. Unlinted code would then cause a failed CI, but that's okey.

[the remark this answers is not in the document]
Comment (StefanieJaeger) [reply package.json:46]: Please remove `gitHooks` and `lint-staged`. And add a linting step to `.github/workflows/build.yml`.

**Review by StefanieJaeger — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (Mafii) [reply package.json:46]: @StefanieJaeger I'll do, thanks so much for the review 🥳

[the remark this answers is not in the document]
Comment (Mafii) [reply package.json]: Addressed

File: README.md (hunk 1 above)
Comment (Mafii) [reply README.md]: 👀

**Comment by Mafii** [comment]
@jeremystucki PRs currently don't run actions, you might have to activate that?

**Comment by jeremystucki** [comment]
There is actually no CI configured except for the deployment. I'll have a look in the next few days and set something up.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -17,6 +17,8 @@ jobs:
         run: npm install
       - name: Build
         run: npm run build
+      - name: Install dependencies
+        run: npm lint
```
Comment (jeremystucki) [inline .github/workflows/build.yml:21]: Wrong name, but it doesn't matter as I already added the check on main. Just update your branch.

Comment (Mafii) [reply .github/workflows/build.yml]: done

**Comment by Mafii** [comment]
Just checked, "works on my machine" ;)

**Comment by jeremystucki** [comment]
@StefanieJaeger please also test locally
@Mafii please resolve all open conversations

Can one of you also test the release build?

**Comment by StefanieJaeger** [comment]
I tested it locally.
Plans, categories, modules and focuses all seem to work.
Selecting a default plan does not save it in cache, you'd first have to change your module selection. I think, default plans should be cached as well. But I have a fix in another PR for that, so no biggie.

**Review by StefanieJaeger — APPROVED** [review]

**Comment by Mafii** [comment]
@jeremystucki can I merge?

**Pull request merged**